### PR TITLE
Add defaults to add_missing_bounds

### LINKED ISSUE #563: [Feature]: Add default value for `axes` argument in `add_missing_bounds()`
### Is your feature request related to a problem?

I think from v0.6.0 `add_missing_bounds` started requiring parameters. 

I liked its previous usage, e.g., `ds.bounds.add_missing_bounds()`, which automatically loops through the Dataset’s axes and attempts to adds bounds to its coordinates if they don’t exist. (By the way, the [documentation](https://xcdat.readthedocs.io/en/latest/generated/xarray.Dataset.bounds.add_missing_bounds.html) is still written in this way). 

### Describe the solution you'd like

I wonder if `add_missing_bounds` can be operated in two optional ways:
(1) `ds.bounds.add_missing_bounds()`: loop through axes and fill missed bounds
(2) `ds.bounds.add_missing_bounds("X")`: fill missed bounds for "X" axis

### Describe alternatives you've considered

_No response_

### Additional context

(added to the post)

Some PMP  and obs4MIP codes have implemented add_missing_bounds() routines but with xcdat update those codes needs changes, and if `ds.bounds.add_missing_bounds()` could be still used, it is going to be very efficient to update xcdat version in PMP dependency. 

@acordonez, @gleckler1, @bosup any thoughts?

### PR DESCRIPTION
**Pull request by acordonez**
## Description

This PR adds default axes to the add_missing_bounds() function. If no axes are provided, the dataset will have X, Y, and T bounds added.

- Closes #563 

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] My changes generate no new warnings
- [x] Any dependent changes have been merged and published in downstream modules

If applicable:

- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass with my changes (locally and CI/CD build)
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have noted that this is a breaking change for a major release (fix or feature that would cause existing functionality to not work as expected)

### CODE REVIEW

**Review by tomvothecoder — APPROVED** [review]
Thank you for addressing this Ana! I had one minor suggestion. Everything else looks good to me.

It looks like the coverage report uploading step in the GitHub Actions build is failing. I'll try to figure out why this is happening. Once the build passes, you can merge.

**Comment by acordonez** [comment]
@tomvothecoder Thanks for your quick review. It look like I don't have write access to merge this.

**Comment by tomvothecoder** [comment]
No problem! I'll merge this PR for you.

I added you as an xCDAT org member with write access. You'll be able to work with the main repo directly instead of using your fork, which should be more convenient next time.

**Comment by tomvothecoder** [comment]
Also FYI that I'll need to do another xCDAT release to have this fix publicly available. I'll look into it within the next two weeks.

If I recall correctly, PCMDI Metrics dev sometimes use the local xCDAT build to get access to the latest fixes/features. It's not recommended, but can be a temporary workaround.

**Comment by lee1043** [comment]
@acordonez thank you for working on this. 

> If I recall correctly, PCMDI Metrics developers sometimes use the local xCDAT build from the main branch to get access to the latest fixes/features. It's not recommended, but can be a temporary workaround.

@tomvothecoder that is true, we can work with the latest version internally but PMP's new version release will be on hold until we get the latest xCDAT with this fix being incorporated. 2 weeks seems fine with us. 

However, I am not sure how obs4MIPs folks are using xCDAT -- they might rely on official releases. As this fix also affects obs4MIPs, I think merging it ASAP would be appreciated. @gleckler1 Did I get this correct?

**Comment by gleckler1** [comment]
obs4MIPs is in high gear, but still prototyping - previously we've modified our codes to deal with past changes to add_missing_bounds but if things are stable now so will be the obs4MIPs processing codes.   I typically get xcdat from conda forge creating an env with it and cmor.  @manaster may want to chime in here..

**Comment by manaster** [comment]
@gleckler1 this is what I do in regards to xCDAT (create and environment with xCDAT from conda forge, so I think that's the official release?).  This proposed fix would seems be great for those of us working on obs4MIPs (at least in terms of our prototype code)!  We were thinking we would have to make some changes to our prototype codes based on changes made to 'add_missing_bounds()' between xCDAT version 0.3.3 and xCDAT version 0.6.0 (the most recent version Peter and I have been using).  The fix proposed here would seem to negate the need for those code changes as far as I can tell.

**Comment by tomvothecoder** [comment]
@lee1043 @gleckler1 @manaster Got it. I'll draft up a changelog for v0.6.1 and see what other small changes can go in it before releasing within the next week or two.

Tagging @xCDAT/core-developers to make everyone aware.

**Pull request merged**
